service/lib/agama/software/manager.rb

```diff
@@ -148,12 +148,15 @@ def select_product(id)
 
         raise ArgumentError unless new_product
 
+        # order is important here as we want to first set internally new product to ensure that cache read will read it properly
+        @product = new_product
+
+        update_repositories(new_product)
+
+        # here it will reset config cache, so do it last to have all stuff in place
```

Comment:
I am not sure why, but even with ordering change it still sometimes does not work and looks like it can still create race conditions when it cache empty product.